### ISSUE TITLE
Can't properly connect or read characteristic from BLE sensor

### ISSUE THREAD

**Issue by mmar58** [Velorexe/Unity-Android-Bluetooth-Low-Energy#49]
Hi, I am was testing this plugin and tried to connect with a BLE sensor. But it's not properly connecting and not reading the characteristic values. I have tested with your temperature characteristic and then tested with Heart rate service UUID and characteristic code, but same result. I have attached the screenshot from device.![Uploading Test BLE 2.jpg…]()

Here is the console log 
[Uploading Error  log.txt…]()

**Comment by mmar58** [comment]
![Uploading Test BLE 1.jpg…]()

**Comment by Velorexe** [comment]
Heya,

Seems like you're trying to connect to a characteristic on a service that isn't on the device (or can't be found). Can you show me the C# code that you're using to read from the characteristic?

**Comment by farunurisonmez** [comment]
<img width="783" alt="Screenshot 2024-05-29 at 23 44 31" src="https://github.com/Velorexe/Unity-Android-Bluetooth-Low-Energy/assets/23061257/bf82232b-1d4e-4d7a-8ec7-c4369fe2c824">

I am getting the same error, @Velorexe 

![Uploading Screenshot 2024-05-30 at 00.29.22.png…]()

**Comment by paulhayes** [comment]
Hi @mmar58, did you resolve your issue?

**Comment by mmar58** [comment]
No, I have paused using this asset and waiting for the fix to resume work
with this asset.

**Comment by paulhayes** [comment]
As we can't reproduce you bug, we can't provide a fix.

Can you try the latest version of the project, using Unity Package Manager and "Add package from git url". If the problem persists, can you try creating an example project, that reproduces the problem, and then zip it up and post it here.

**Comment by mmar58** [comment]
Hi @paulhayes , thanks for considering about this. I will test the latest project and give you feedback about this and if this problem persist, I will share the demo, so that you can test it too.

**Comment by SaaskunDevs** [comment]
Hi! I have the same problem, triying to connect a LED strip.
I just import the repostory and test the sample scene.

The error ocurse before the SubscribeToExampleService function

**Comment by paulhayes** [comment]
Hi @SaaskunDevs which version of the project are you using? Are you importing via Package Manager or the old unity package file under releases?

**Comment by SaaskunDevs** [comment]
> Hi @SaaskunDevs which version of the project are you using? Are you importing via Package Manager or the old unity package file under releases?

I'm Using the Package Manager

**Comment by paulhayes** [comment]
Great

Can you provide Unity Version, android SDK, so I can try and reproduce the issue.
